pkg/logs/launchers/listener/udp.go

```diff
@@ -86,15 +86,16 @@ func (l *UDPListener) newUDPConnection() (net.Conn, error) {
 }
 
 // read reads data from the tailer connection, returns an error if it failed and reset the tailer.
-func (l *UDPListener) read(tailer *tailer.Tailer) ([]byte, error) {
+func (l *UDPListener) read(tailer *tailer.Tailer) ([]byte, string, error) {
 	frame := make([]byte, l.frameSize+1)
-	n, err := tailer.Conn.Read(frame)
+	// Add casting to UDPConn
+	n, udpAddr, err := tailer.Conn.(*net.UDPConn).ReadFromUDP(frame)
```

Comment:
it turns out the connection is created in this file too! 
[ListenUDP](https://github.com/DataDog/datadog-agent/blob/456dcfdc2af470855412daece4595400808f2ea6/pkg/logs/launchers/listener/udp.go#L85) returns a `UDPConn` instance. you can store this on the [UDPListener](https://github.com/DataDog/datadog-agent/blob/456dcfdc2af470855412daece4595400808f2ea6/pkg/logs/launchers/listener/udp.go#L31) struct. 

And then on this line, you can replace it with `l.Conn.ReadFromUDP(frame)` and avoid the type cast :)